src/math/src/ed25519.cairo

```diff
@@ -261,33 +290,44 @@ fn check_group_equation(
     lhs == rhs
 }
 
-fn verify_signature(msg: Span<u8>, signature: Span<u8>, pub_key: Span<u8>) -> bool {
-    if (pub_key.len() != PUB_KEY_LEN || signature.len() != SIG_LEN) {
+fn verify_signature(msg: Span<u8>, signature: Span<u256>, pub_key: u256) -> bool {
+    let r: u256 = *signature.get(0).unwrap().unbox();
```

Comment:
why not use `IndexView` directly ?

Reply:
Can't find the implementation, do I have to implement it or is it somewhere already?

Reply:
https://github.com/starkware-libs/cairo/blob/c064d5e789d5bf194691eca9bd0f3b508caf83fc/corelib/src/array.cairo#L218-L224
this would change the code to
```suggestion
let r: u256 = *signature[0];